csvs_update_sqlite/cli.py

```diff
@@ -284,23 +292,29 @@ def cli(
                 add_index(conn, df.table_name, index_defn)
 
     # Create FTS tables
-    if fts:
+    if fts or fts_regex:
+        # if fts_regex: click.echo(fts_regex)
         fts_version = best_fts_version()
         if not fts_version:
             conn.close()
             raise click.BadParameter(
                 "Your SQLite version does not support any variant of FTS"
-            )
+            ) 
         # Check that columns make sense
         for table, df in created_tables.items():
-            for fts_column in fts:
-                if fts_column not in df.columns:
-                    raise click.BadParameter(
-                        'FTS column "{}" does not exist'.format(fts_column)
-                    )
-
-        generate_and_populate_fts(conn, created_tables.keys(), fts, foreign_keys)
-
+            if fts_regex:
+                for col in df:
+                    if re.match(fts_regex, col, re.IGNORECASE):
+                        fts = (*fts,col,) if fts else (col,)
+            if fts:
+                for fts_column in fts:
+                    if fts_column not in df.columns:
+                        raise click.BadParameter(
+                            'FTS column "{}" does not exist'.format(fts_column)
+                        )
+            
+            generate_and_populate_fts(conn, created_tables.keys(), fts, foreign_keys)
```

Comment:
should this be outside of the for loop?

Reply:
I don't think so. The point of this block is to gather up gather eligible column names in a tuple and add full-text search to them. But there is one error that needs resolving: If an FTS table already exists from a previous run of `csvs-update-sqlite` with `-fr` flag and user runs `csvs-update-sqlite` with an`-fr` string to try to add or change columns that have FTS, the script will run into an error trying to create an FTS table because it already exists. Still, I think this fork can be useful to your users. But I leave it to you to decide.